File: app/src/main/java/com/example/whattowatch/model/mymodel/MyMovieModel.java
package com.example.whattowatch.model.mymodel;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class MyMovieModel {
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @PrimaryKey(autoGenerate = true)
    private Integer id;
    private  String mTitle;
    private  String mOverview;
    private  String mDate;
    private  Float mRate;
    private  String mImageLink;
    private Integer movieId;
    private  String type;



    public MyMovieModel(String title, String overview, String date, Float rate, String imageLink, Integer idMovie, String type) {
        mTitle = title;
        mOverview = overview;
        mDate = date;
        mRate = rate;
        mImageLink = imageLink;
        movieId = idMovie;
        this.type = type;
    }
    public MyMovieModel(){}

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Integer getMovieId() {
        return movieId;
    }

    public void setMovieId(Integer movieId) {
        this.movieId = movieId;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public String getOverview() {
        return mOverview;
    }

    public void setOverview(String overview) {
        mOverview = overview;
    }

    public String getDate() {
        return mDate;
    }

    public void setDate(String date) {
        mDate = date;
    }

    public Float getRate() {
        return mRate;
    }

    public void setRate(Float rate) {
        mRate = rate;
    }

    public String getImageLink() {
        return mImageLink;
    }

    public void setImageLink(String imageLink) {
        mImageLink = imageLink;
    }


}

File: app/src/main/java/com/example/whattowatch/model/mymodel/MyMovieModel.kt
package com.example.whattowatch.model.mymodel

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class MyMovieModel {

    @PrimaryKey(autoGenerate = true)
    var id: Int? = null
    var title: String? = null
    var overview: String? = null
    var date: String? = null
    var rate: Float? = null
    var imageLink: String? = null
    var movieId: Int? = null
    var type: String? = null

    constructor(title: String?, overview: String?, date: String?, rate: Float?, imageLink: String?, idMovie: Int?, type: String?) {
        this.title = title
        this.overview = overview
        this.date = date
        this.rate = rate
        this.imageLink = imageLink
        movieId = idMovie
        this.type = type
    }

    constructor() {}

}